File: adt/adt_all_20250123_3/e/main.py
# https://atcoder.jp/contests/abc227/tasks/abc227_c
# 最大ケースだと、自分のPCで約6秒。意味が分かりません

N = int(input())

ans = 0

A = 1
while A ** 3 <= N:
    B = A
    while A * (B ** 2) <= N:
        max_c = int(N / (A * B))
        ans += max_c - B + 1
        B += 1

    A += 1

print(ans)

File: adt/adt_all_20250123_3/e/main.cpp
// https://atcoder.jp/contests/abc227/tasks/abc227_c
// 初めてC++を自力で書けた！！

#include <bits/stdc++.h>
using namespace std;

int main() {
    long long N;
    cin >> N;

    long long ans = 0;
    long long A = 1;
    long long B;
    long long max_c;
    while (A * A * A <= N)
    {
        B = A;
        while (A * B * B <= N)
        {
            max_c = N / (A * B);
            ans += max_c - B + 1;
            B += 1;
        }

        A += 1;
    }
    cout << ans;

}
